app/code/Magento/Customer/Controller/Address/FormPost.php

```diff
@@ -89,18 +96,44 @@ public function __construct(
             $dataProcessor,
             $dataObjectHelper,
             $resultForwardFactory,
-            $resultPageFactory
+            $resultPageFactory,
+            $customerRepository
         );
     }
 
     /**
      * Extract address from request
      *
-     * @return \Magento\Customer\Api\Data\AddressInterface
+     * @return AddressInterface
+     * @throws \Exception
+     * @deprecated Use extractCurrentAddress instead
+     */
+    protected function _extractAddress(): AddressInterface
+    {
+        return $this->extractCurrentAddress();
+    }
+    /**
+     * Retrieve existing address data
+     *
+     * @return array
+     * @throws \Exception
+     * @deprecated Use _getExistingAddressData instead
+     */
+    protected function getExistingAddressData(): array
+    {
+        return $this->_getExistingAddressData();
+    }
+
+    /**
+     * Extract address from request
+     *
+     * @param CustomerInterface|null $customer
+     * @return AddressInterface
+     * @throws \Exception
      */
-    protected function _extractAddress()
+    protected function extractCurrentAddress(CustomerInterface $customer = null): AddressInterface
```

Comment:
Unfortunately, we can't change public and protected methods signature according to our [Backward Compatible Development Guide](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/)

Reply:
there is nothing changed, the original `protected function _extractAddress()` is saved, it begins from 111 line. The `extractCurrentAddress` method is a new one.